### PR TITLE
Feature/detailed filters

### PR DESCRIPTION
**Pull request by JanPSchwarz**
please note: 
- in contrast to the US the searchBar component wasn't changed and the logic regarding the filter functions was kept on the emotion-records-page due to the fact that the logic is mostly passed  to and used in the EmotionsList component which is rendered there
- there is `exampleData.js` (approx 50 lines of code) added in the utils directory which is only for testing and not particulary part of the US. The exampleDate stores one entry per day for 90 days starting from yesterday in the localStorage to test the filter functions accordingly

log: 
- [React-Day-Picker](https://react-day-picker.js.org/using-daypicker/styling) was used as a library and implemented
- core changes are in `emotion-records.js `(300 lines added)
- `EmotionRecordsList` was updated to display visuals accordingly to filterFunctioins
- `emotionEntries` object-structure was changed to have an additional date `key` for comparisons in filter functions
- some minor bugs and fixed were made along the way

### CODE REVIEW

File: pages/emotion-records.js [hunk 1]
```diff
@@ -84,50 +249,116 @@ export default function EmotionRecords({
...
+      month: "2-digit",
+      year: "numeric",
+    }).format(selectedDate);
+
+    return date;
+  }
+
   return (
     <StyledFlexColumnWrapper>
       <StyledPageHeader>
```
Comment (benediktbrenk) [inline pages/emotion-records.js:270]: The name StyledPageHeader seems a bit misleading to me. You already have a Header component in the layout. The name (Styled)FilterSection seems more appropriate to me from what code it contains.

**Review by benediktbrenk — CHANGES_REQUESTED** [review]
The functionality and logic is convincing! 

I've some regards concerning structure. As a suggestion: You could use components in seperate files (Filter/index.js + Filter/Filter.styled.js) if parts of the code belong together as a logical unit. This will improve orientation and readability in your emotion-records.js code enormously.

-> Create a component for the calendar modal
-> Create a component for the filter section

In any case: Please change the name of the StyledPageHeader (l.99) to a more functional naming.

**Comment by JanPSchwarz** [comment]
> The functionality and logic is convincing!
> 
> I've some regards concerning structure. As a suggestion: You could use components in seperate files (Filter/index.js + Filter/Filter.styled.js) if parts of the code belong together as a logical unit. This will improve orientation and readability in your emotion-records.js code enormously.
> 
> -> Create a component for the calendar modal -> Create a component for the filter section
> 
> In any case: Please change the name of the StyledPageHeader (l.99) to a more functional naming.

Thanks for your review! I will for sure change the StyledPageHeader according to your suggestion and i will consider refactoring to smaller components

**Comment by JanPSchwarz** [comment]
After last PR i decided to restructure a bit more and changed the `searchBar` to be `FilterEmotionEntries` which holds all filter functions and the `DayPicker` as a separate component

**Review by benediktbrenk — APPROVED** [review]

**Pull request merged**